internal/pkg/manifest/lb_web_svc.go

```diff
@@ -99,16 +105,22 @@ func newDefaultLoadBalancedWebService() *LoadBalancedWebService {
 // MarshalBinary serializes the manifest object into a binary YAML document.
 // Implements the encoding.BinaryMarshaler interface.
 func (s *LoadBalancedWebService) MarshalBinary() ([]byte, error) {
-	content, err := s.parser.Parse(lbWebSvcManifestPath, *s)
+	content, err := s.parser.Parse(lbWebSvcManifestPath, *s, template.WithFuncs(map[string]interface{}{
+		"dirName": tplDirName,
+	}))
 	if err != nil {
 		return nil, err
 	}
 	return content.Bytes(), nil
 }
 
-// DockerfilePath returns the image build path.
-func (s *LoadBalancedWebService) DockerfilePath() string {
-	return aws.StringValue(s.Image.Build)
+func tplDirName(s string) string {
+	return filepath.Dir(s)
+}
+
+//BuildArgs returns a docker.BuildArguments object given a ws root directory.
+func (s *LoadBalancedWebService) BuildArgs(wsRoot string) *DockerBuildArgs {
```

Comment:
Can we add this method to `Service` struct, so we don't always have to redefine it for every pattern? Or is it not possible (no prob if it's not, just curioust)

Reply:
I tried but it's weird, because the method needs access to the Image member of the LoadBalancedWebService or BackendService struct, but I don't know how to cast a particular member of an unmarshaled struct to an interface instead of just the whole struct